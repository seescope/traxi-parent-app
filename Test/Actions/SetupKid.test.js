import setupKid from '../../App/Actions/SetupKid';
import mockFirebase, { mockSet } from 'firebase';

global.Promise = require.requireActual('promise');

it('sets up a Kid', () => {
  const TEST_KID = {
    name: 'Test Name',
    phoneNumber: '+61401633346',
  };

  Math.random = () => 0.1234;

  const mockDispatch = jest.fn();
  const mockGetState = () => ({
    selectedKid: TEST_KID,
  });

  return setupKid()(mockDispatch, mockGetState).then(() => {
    // Dispatch SET_UUID
    expect(mockDispatch.mock.calls[0]).toMatchSnapshot();

    // Create Firebase refs with correct path
    expect(mockFirebase.mock.calls[0]).toMatchSnapshot();
    expect(mockFirebase.mock.calls[1]).toMatchSnapshot();

    // Post setup ID to Firebase
    expect(mockSet.mock.calls[0]).toMatchSnapshot();

    // Post kid to Firebase
    expect(mockSet.mock.calls[1]).toMatchSnapshot();
  });
});
