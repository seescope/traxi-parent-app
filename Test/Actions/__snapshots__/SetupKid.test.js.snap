exports[`test sets up a Kid 1`] = `
Array [
  Object {
    "UUID": "non-random-uuid",
    "setupID": 1234,
    "type": "ADD_KID",
  },
]
`;

exports[`test sets up a Kid 2`] = `
Array [
  "https://traxiapp.firebaseio.com/setupIDs/1234",
]
`;

exports[`test sets up a Kid 3`] = `
Array [
  "https://traxiapp.firebaseio.com/kids/non-random-uuid",
]
`;

exports[`test sets up a Kid 4`] = `
Array [
  "non-random-uuid",
  [Function],
]
`;

exports[`test sets up a Kid 5`] = `
Array [
  Object {
    "UUID": "non-random-uuid",
    "deviceType": "unknown",
    "name": "Test Name",
    "phoneNumber": "+61401633346",
    "setupID": 1234,
    "status": "WAITING",
  },
  [Function],
]
`;
